### PR TITLE
Update thin-egress-app documentatation re: CUMULUS-1958

### PR DESCRIPTION
**Pull request by Jkovarik**
**Summary:** Summary of changes

Addresses [CUMULUS-1958](https://bugs.earthdata.nasa.gov/browse/CUMULUS-1958)

## Changes

Documentation updates for CUMULUS-1958

### CODE REVIEW

File: docs/deployment/thin-egress-app.md [hunk 1]
```diff
@@ -8,16 +8,24 @@ hide_title: true
 
 The [Thin Egress App (TEA)](https://github.com/asfadmin/thin-egress-app) is an app running in Lambda that allows retrieving data from S3 using temporary links and provides URS integration.
 
-Note: If you are using the `cumulus` module, you will not need the information on this page, as Cumulus packages TEA within its `distribution` tf-module, which is pre-configured and included in our releases as a submodule of the `cumulus` module. However, if you are using individual Cumulus modules in your own Terraform configuration, or wish to configure TEA on your own, the information below is important to set up distribution.
+## Cumulus "Packaged" Deployment
 
-## Thin Egress App deployment
+ If you are using the `cumulus` module or the core `distribution` module (deployed with the `cumulus` module), you will not need to manually deploy TEA as Cumulus packages TEA within its `distribution` tf-module, which is pre-configured and included in our releases as a submodule of the `cumulus` module.
```
Comment (yjpa7145) [inline docs/deployment/thin-egress-app.md:13]:
```markdown
If you are using the `cumulus` module or the core `distribution` module, you will not need to manually deploy TEA. Cumulus packages TEA within its `distribution` tf-module, which is pre-configured and included in our releases.

By default, Core provides a one-to-one mapping where each protected and public bucket map to a path of the same name as the bucket.

The `cumulus` and `distribution` modules provide a `bucket_map_key` input variable which allows you to specify the path to a custom [bucket map YAML](https://github.com/asfadmin/thin-egress-app#buckets-and-bucket-map).

**Note: Cumulus only supports a one-to-one mapping of bucket->path name. Mapping multiple paths to the same bucket will result in a deployment failure.**
```

File: docs/deployment/thin-egress-app.md [hunk 2]
```diff
@@ -105,3 +113,17 @@ Pass `api_distribution_url` to your `archive` module's `distribution_url` var.
 
 You will also need to configure the `tea_urs_redirect_uri` value as a Redirect
 URI in your app's URS configuration.
+
+### Cumulus Configuration
+
+The default Cumulus module generates a file at `s3://${system_bucket}/distribution_bucket_map.json` -- if you've manually deployed TEA outside the `cumulus` module, you'll need to provide this configuration file as part of your Terraform deployment if you wish for Cumulus's CMR related functionality (`Move Granules` task and all CMR reconciliation functions) to function.
```
Comment (yjpa7145) [inline docs/deployment/thin-egress-app.md:119]:
```suggestion
The default Cumulus module generates a file at `s3://${system_bucket}/distribution_bucket_map.json`. If you've manually deployed TEA outside the `cumulus` module, you'll need to provide this configuration file as part of your Terraform deployment.
```

File: docs/deployment/thin-egress-app.md (hunk 1 above)
Comment (Jkovarik) [reply docs/deployment/thin-egress-app.md:13]: 👍

File: docs/deployment/thin-egress-app.md [hunk 3]
```diff
@@ -8,16 +8,26 @@ hide_title: true
...
 
-Note: If you are using the `cumulus` module, you will not need the information on this page, as Cumulus packages TEA within its `distribution` tf-module, which is pre-configured and included in our releases as a submodule of the `cumulus` module. However, if you are using individual Cumulus modules in your own Terraform configuration, or wish to configure TEA on your own, the information below is important to set up distribution.
+## Cumulus "Packaged" Deployment
 
-## Thin Egress App deployment
+If you are using the `cumulus` module or the core `distribution` module, you will not need to manually deploy TEA. Cumulus packages TEA within its `distribution` tf-module, which is pre-configured and included in our releases.
+
+By default, Core provides a one-to-one mapping where each protected and public bucket map to a path of the same name as the bucket.
+
+The `cumulus` and `distribution` modules provide a `bucket_map_key` input variable which allows you to specify the path to a custom [bucket map YAML](https://github.com/asfadmin/thin-egress-app#buckets-and-bucket-map).
```
Comment (laurenfrederick) [inline docs/deployment/thin-egress-app.md:17]: When we say we're providing the file, is this a local path or does the bucket map need to be in S3 and it's the S3 location?

File: docs/deployment/thin-egress-app.md [hunk 4]
```diff
@@ -105,3 +115,17 @@ Pass `api_distribution_url` to your `archive` module's `distribution_url` var.
...
+
+### Cumulus Configuration
+
+The default Cumulus module generates a file at `s3://${system_bucket}/distribution_bucket_map.json`. If you've manually deployed TEA outside the `cumulus` module, you'll need to provide this configuration file as part of your Terraform deployment.
+
+The configuration file is a simple json mapping of the form:
+
+```json
+{
+  "bucket_name": "bucket_path_in_TEA"
```
Comment (laurenfrederick) [inline docs/deployment/thin-egress-app.md:127]: I think we need more of an example here like:

`"protected": "MOD09GQ/protectedData/"` whatever makes sense here

File: docs/deployment/thin-egress-app.md (hunk 3 above)
Comment (Jkovarik) [reply docs/deployment/thin-egress-app.md:17]: Good question - we should probably clarify we're talking about the system bucket here.  This is a pass through for https://github.com/asfadmin/thin-egress-app/blob/devel/terraform/variables.tf#L7

Our current config being passed from the distribution module is: 

  bucket_map_file                    = var.bucket_map_key == null ? aws_s3_bucket_object.bucket_map_yaml[0].key : var.bucket_map_key
  config_bucket                      = var.system_bucket

File: docs/deployment/thin-egress-app.md (hunk 4 above)
Comment (Jkovarik) [reply docs/deployment/thin-egress-app.md:127]: 👍  I'll add a more concrete example

Comment (Jkovarik) [reply docs/deployment/thin-egress-app.md]: fb4864cd5

File: docs/deployment/thin-egress-app.md (hunk 3 above)
Comment (Jkovarik) [reply docs/deployment/thin-egress-app.md]: fb4864cd5

**Comment by Jkovarik** [comment]
@laurenfrederick  Thanks for the review - I think I've addressed your concerns, let me know what you think!

**Pull request merged**